### DISCUSSION TITLE
Comment ajouter l'outil "Ajouter des données" ?

### DISCUSSION THREAD

**Discussion by Heshzer** [mviewer/mviewer#901] [Q&A]
Bonjour à tous,

Je travaille actuellement sur un projet utilisant Mviewer couplé à un GeoServer pour la gestion et la visualisation de données. 
J'aimerais permettre aux utilisateurs de notre plateforme d'ajouter des couches de données directement à partir des couches enregistrés sur notre GeoServer, de manière autonome, en utilisant un outil similaire à l'outil "Ajouter des données" que l'on peut voir sur la carte suivante : https://www.geo2france.fr/mviewer/?config=apps/pcrs.xml#

L'objectif est de permettre à mes utilisateurs d'accéder à d'autres couches disponibles sur notre GeoServer et de les afficher sur la carte en cours pour leurs propres analyses.

Je suis à la recherche de conseils ou de documentation sur la manière d'intégrer cet outil dans mon instance de Mviewer.

    

1. Faut-il configurer un plugin spécifique ?
2. Y a-t-il des paramètres particuliers à ajuster dans le fichier de configuration XML ou ailleurs ?
Des exemples de configurations similaires ou de projets qui ont intégré cette fonctionnalité seraient très appréciés.

Merci d'avance pour votre aide !

**Comment by Gaetanbrl** [comment]
Bonjour  @Heshzer,

La documentation est disponible en ligne pour ajouter le panneau d'ajout des données via le paramètre `addlayerstools` :
https://mviewerdoc.readthedocs.io/fr/stable/doc_tech/config_app.html

Avec cet outil, vous pouvez utiliser un fichier .json de configuration qu'il faudra adapter selon les flux que vous souhaitez proposer à vos utilisateurs :
https://github.com/mviewer/mviewer/blob/master/data/ogc_csw_server.json

Un exemple est disponible ici :
https://www.geo2france.fr/mviewer/?config=demo/addlayers.xml#

Voici le code source XML de cet exemple :
https://github.com/mviewer/mviewer/blob/master/demo/addlayers.xml


En bref : 

- Rajouter `addlayerstools=true` dans la balise `<application> ` de votre configuration XML
- Modifier le fichier `data/ogc_csw_server.json` pour y ajouter vos services OGC ou CSW

**Reply by Heshzer** [reply]
Bonjour @Gaetanbrl,

Merci beaucoup pour ces indications, c'est super !
